cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java

```diff
@@ -182,4 +183,23 @@ private String getRequestingUser() throws IllegalArgumentException {
     LOG.trace("Got requesting principal {}", principal);
     return principal.getName();
   }
+
+  private String getDotRoleName(EntityId entityId, Principal principal) {
+    String principalType;
+    // use a single character for principal type, to reduce role name length
+    switch (principal.getType()) {
+      case USER:
+        principalType = "u";
+        break;
+      case GROUP:
+        principalType = "g";
+        break;
+      case ROLE:
+        principalType = "r";
+        break;
+      default:
+        throw new IllegalStateException("Invalid principal type " + principal.getType());
+    }
+    return Joiner.on(ENTITY_ROLE_PREFIX).join("", entityId.toString(), principal.getName(), principalType);
```

Comment:
Why do we need "" ?
Also, get we just get the first character from .name().toLowerCase().charAt(0) and get rid of the switch case.